core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/HashJoinIteration.java

```diff
@@ -87,6 +87,20 @@ public HashJoinIteration(EvaluationStrategy strategy, TupleExpr left, TupleExpr
 
 		this.leftJoin = leftJoin;
 	}
+	
```

Comment:
It might be a good idea if the constructor above delegates to this new constructor with a this(...,strategy.evaluate(left, bindings),...).  You could also make it take CloseableIteration, Set<String> for both left and right to maintain a bit of consistency with the param types.